### PR TITLE
Bug 1306099 - New console frontend: Fix scrolling

### LINKED ISSUE #255: Entering a command should automatically scroll the output to the bottom
When you run a command, the console output should automatically scroll to where the command is displayed (to the bottom)

### LINKED ISSUE #315: Small UX nit: expanding messages may drop you to the bottom
This only happens if you haven't reached the bottom of the screen yet (I think?). If I have a few messages, with one that has expandable data, like this:

screen shot 2016-09-27 at 11 58 22 am

If I expand it, and it's really long, it drops me to the bottom of the expanded output:

screen shot 2016-09-27 at 11 58 33 am

Here's the previous console behavior:

screen shot 2016-09-27 at 11 58 46 am

screen shot 2016-09-27 at 11 58 55 am

### PR DESCRIPTION
**Pull request by linclark**
Fixes #255, fixes #315 

@bgrins, do you have time for review?
## Testing instructions
1. You can test that it scrolls to the bottom of cached messages with 

```
data:text/html;charset=utf-8,<p>Test keyboard accessibility</p>
  <script>
    for (let i = 1; i <= 100; i++) {
      console.log("console message " + i);
    }
  </script>
```
1. After that you can scroll to the top and then enter a command. This fixes #255
2. Then add a `console.trace()` call. Close it and scroll so that the area that would expand is hidden. Then open again. It won't push the scroll down. This fixes #315.

### CODE REVIEW

**Comment by linclark** [comment]
I just realized we probably only want to scroll to the bottom in the case of console commands and evaluation results, which would require a change in the reducer here.

**Comment by linclark** [comment]
@bgrins this is ready for review

File: devtools/client/webconsole/new-console-output/components/console-output.js [hunk 1]
```diff
@@ -27,25 +28,33 @@ const ConsoleOutput = createClass({
     onViewSourceInDebugger: PropTypes.func.isRequired,
     openNetworkPanel: PropTypes.func.isRequired,
     openLink: PropTypes.func.isRequired,
+    autoscroll: PropTypes.object.isRequired,
```
Comment (bgrins) [inline devtools/client/webconsole/new-console-output/components/console-output.js]: Should this be a bool?

File: devtools/client/webconsole/new-console-output/components/console-output.js [hunk 2]
```diff
@@ -69,15 +78,28 @@ const ConsoleOutput = createClass({
...
+          autoscroll,
         })
       );
     });
     return (
-      dom.div({className: "webconsole-output"}, messageNodes)
+      dom.div({
+        className: "webconsole-output",
+        ref: node => {
+          if (node !== null) {
```
Comment (bgrins) [inline devtools/client/webconsole/new-console-output/components/console-output.js]: When would node ever be null in a ref callback?

File: devtools/client/webconsole/new-console-output/components/console-output.js (hunk 1 above)
Comment (linclark) [reply devtools/client/webconsole/new-console-output/components/console-output.js]: Yes, thanks for catching. I had more complicated state/logic before, but then realized I could simplify.

**Pull request merged**